dla_cnn/desi/training_sets.py

```diff
@@ -37,12 +37,13 @@ def split_sightline_into_samples(sightline, REST_RANGE, kernel=400):
     lam, lam_rest, ix_dla_range = get_lam_data(sightline.loglam, sightline.z_qso, REST_RANGE)
     #samplerangepx = int(kernel*pos_sample_kernel_percent/2) #60
     kernelrangepx = int(kernel/2) #200
+    cut=((np.nonzero(ix_dla_range)[0])>=kernelrangepx)&((np.nonzero(ix_dla_range)[0])<=(len(lam)-kernelrangepx-1))
```

Comment:
Can you explain what `cut` does?  Thanks.

Reply:
If np.nonzero(ix_dla_range)[0]<kernel/2 or np.nonzero(ix_dla_range)[0]>len(lam)-kernelrangepx-1)，the code can not generate a kernel size snippet.
 Such as np.amin(np.nonzero(ix_dla_range))=0,then when split sightline into samples:[0-200,0+200],will not work.

Reply:
I think I understand, but give a real example, ie. from one of the mock spectra.
Thanks.

Reply:
when run  this module without cut ,you can see this error:

ValueError                                Traceback (most recent call last)
<ipython-input-5-414e71b5a61c> in <module>
      1 label_sightline(sightline, kernel=400, REST_RANGE=[900,1346], pos_sample_kernel_percent=0.3)#update
----> 2 data_split=split_sightline_into_samples(sightline)#update
      3 sample_masks=select_samples_50p_pos_neg(sightline)#update

~/dla_cnn/desi/training_sets.py in split_sightline_into_samples(sightline, REST_RANGE, kernel)
     45 
     46     # FLUXES - Produce a 1748x400 matrix of flux values
---> 47     fluxes_matrix = np.vstack(map(lambda x:x[0][x[1]-kernelrangepx:x[1]+kernelrangepx],zip(itertools.repeat(sightline.flux), np.nonzero(ix_dla_range)[0])))
     48     #fluxes_matrix = np.vstack(map(lambda x:x[0][x[1]-kernelrangepx:x[1]+kernelrangepx],zip(itertools.repeat(sightline.flux), np.nonzero(ix_dla_range)[0][cut])))
     49     # Return

<__array_function__ internals> in vstack(*args, **kwargs)

~/opt/anaconda3/lib/python3.7/site-packages/numpy/core/shape_base.py in vstack(tup)
    280     if not isinstance(arrs, list):
    281         arrs = [arrs]
--> 282     return _nx.concatenate(arrs, 0)
    283 
    284 

<__array_function__ internals> in concatenate(*args, **kwargs)

ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 0 and the array at index 200 has size 400

Reply:
I fear I still don't understand.  I suggest you:

- Load a Sightline in a Notebook
- Plot it
- Run the code
- Explain why it crashes the code

Reply:
ok, I will do this